projects/driver_controls/inc/pedal_calibration.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+// This module provides the means for pedal calibration.
+// Throttle module takes in a structure, ThrottleCalibrationData, that is calibrated in this module.
+//
+// Calibration has basically 3 stages. First stage is about reading and recording pedal inputs from
+// pedal at full brake position. The second stage is same as first but for full throttle position.
+// The last stage is about calculating the results into a ThrottleCalibrationData structure that
+// throttle module uses.
+//
+// The procedure is set up in test_pedal_calibration with the help of a FSM that
+// when switched to a state calls the corresponding functions from this module. Look at
+// pedal_calibration_fsm module for more information.
+#include "ads1015.h"
+#include "throttle.h"
+
+// Pedal takes 2 ADC channels.
+// Calibration determines the high and the low resolution channel.
+typedef enum {
+  PEDAL_CALIBRATION_CHANNEL_A = 0,
+  PEDAL_CALIBRATION_CHANNEL_B,
+  NUM_PEDAL_CALIBRATION_CHANNELS
+} PedalCalibrationChannel;
+
+// Pedal inputs are read in 2 positions: full throttle and full brake.
+typedef enum {
+  PEDAL_CALIBRATION_STATE_FULL_BRAKE = 0,  // Fully pressed.
+  PEDAL_CALIBRATION_STATE_FULL_THROTTLE,   // Fully depressed.
+  NUM_PEDAL_CALIBRATION_STATES
+} PedalCalibrationState;
+
+// The range is the structure that holds max and min ADC readings on a channel in a fixed state.
+// For example, reading the pedal at full brake position would give a range of values, not one.
+typedef struct PedalCalibrationRange {
+  int16_t min;
+  int16_t max;
+} PedalCalibrationRange;
+
+typedef struct PedalCalibrationSettings {
```

Comment:
There should probably be some description of the math underlying this calibration here. I shouldn't have to look at the implementation to understand what `bounds_tolerance` means. As it is it is somewhat ambiguous.